### DISCUSSION TITLE
Filter a list

### DISCUSSION THREAD

**Discussion by davidhunter08** [nhsuk/nhsapp-frontend#360] [Experimental: Components and patterns]
## What

Users can filter items in a list.

## Related to

- https://github.com/nhsuk/nhsapp-frontend/issues/31
- https://github.com/nhsuk/nhsapp-frontend/issues/30
- https://github.com/nhsuk/nhsapp-frontend/issues/58
- https://github.com/nhsuk/nhsapp-frontend/issues/248

## More info

- https://github.com/nhsuk/nhsuk-service-manual-community-backlog/issues/411
- https://github.com/alphagov/govuk-design-system-backlog/issues/133

**Comment by davidhunter08** [comment]
<img width="374" alt="Screenshot 2024-04-23 at 11 41 20" src="https://github.com/nhsuk/nhsapp-frontend/assets/14331000/72997c33-5b7a-4ce8-ba89-fed571a7cdd7">

Notification & messaging

**Comment by anna-rigo** [comment]
From two rounds of moderated usability research for document and questionnaire management, the secondary care team did not identify any critical usability issues. 

The main insights gathered were more on the noticeability of the filter/sort function as majority of participants did not notice filter by and sort by unless they were prompted. 
- This could be due to constraints of setting tasks during research and the attention of participants. 
- Other insights included users appreciation for being able to breakdown overwhelming amounts of information to get to what they need when they need it. Especially for those with long medical histories and those who manage multiple ongoing conditions and therefore would have a lot of documents to navigate through. 
- Some users suggested additional filters including by specialist, condition and date to further improve their experience with filters.

Users in these research rounds were able to:
Filter by:
* Document vs questionnaire
* Action needed vs no action needed

And sort by:
* Newest vs oldest vs alphabetical

These are existing functionalities within messages in the NHS App, adapted and brought into this screen.


![Image](https://github.com/user-attachments/assets/a087cb3f-e6fb-4643-af99-6c8d8daf9701)

**Comment by michaelgallagher** [comment]
@anna-rigo was there any insight into whether the noticeability was related to the UI of the "button"? it doesn't have a very buttony appearance, so i am wondering if users have effectively been trained to "know what a button looks like" in an nhs context, and thus don't see the filter "button" as meeting the heuristic criteria for "oh, that's a thing i can click".

**Comment by joejulier-nhs** [comment]
It would also be really helpful to know how many items were in the list participants could filter. Your comment mentions participants highlighted the use case for filters being when there are overwhelming amounts of information to work through, so maybe if the prototype included a relatively manageable amount of information, participants simply didn't think to look for a filter because they didn't feel they needed it? 

I did some work on filters for an internal facing service a while back and found users often had quite a high tolerance for scrolling and scanning before resorting to using filters - in fact, iirc, they basically only used the filter once they realised it would be impossible to complete the task through scanning/scrolling.

**Comment by anna-rigo** [comment]
> [@anna-rigo](https://github.com/anna-rigo) was there any insight into whether the noticeability was related to the UI of the "button"? it doesn't have a very buttony appearance, so i am wondering if users have effectively been trained to "know what a button looks like" in an nhs context, and thus don't see the filter "button" as meeting the heuristic criteria for "oh, that's a thing i can click".

That's a very good point, Mike. I do not believe there was any insights on the noticeability being related to the UI, I will double check with the secondary care team as they did the research and report back. However, it could be assumed that might be the case based on the other user insights we see emerge from other research by other teams, which will also be added here soon.

**Comment by anna-rigo** [comment]
The Notification and Messaging team had done some usability testing in July 2023 to better understand how filtering and sorting could navigate users with their messages on the NHS App. As currently users do not have any other functionality to enable them to manage their messages from health and care providers in the way that best supports them being able to find, organise and refer back to their messages which could contain important information. 

Key findings: 
1. ﻿﻿﻿All users tended to scroll rather than use filter or sort features initially due to it being an intuitive behaviour.
2. ﻿﻿﻿Both filtering and sort were highly usable, with 6/6 users able to apply a filter and 5/6 able to apply a sort.
3. ﻿﻿﻿Filtering was perceived as more practical and useful than sorting due to the categories available (e.g. sender over alphabetically) and ability to reduce the number of messages in the inbox.
4. ﻿﻿﻿Filtering met users’ expectations but they expect to be able to filter by multiple options and additional filters could be useful in future (e.g., by message type, topic and urgency).
5. ﻿﻿﻿Other inbox management functionalities were familiar to users, including removing messages (delete / archive) and search.

Quotes
"Not necessary to have any of the sort functions with the filter. I wouldn't actually bother with the sort" 
"Filter is also more practical and easier to use"
" I didn't find use for sort other than unread part"


<img width="302" alt="Screenshot" src="https://github.com/user-attachments/assets/4f263d11-47c6-4a35-90ff-46a0ad146a94">

**Comment by anna-rigo** [comment]
The GP services team completed some usability testing in May 2024 to test implications of design changes on user experience for GP appointments booking. The objectives included: (1) to validate if users can easily complete core journeys and (2) to identify opportunities for improvements and further iterations. Overall, users understood how to use the filter but thought filter features could be improved.

**Key findings:** 
- Users recognised that if they don't have a location preference, they can leave the box unticked so we can remove the 'no preference option' and free up space on the page 
- Users would like to filter by date
- At first most participants scrolled through all appointments to find the 29th (they were asked to book an appointment in 3 weeks time) rather than trying to filter
- Only a few users noticed and used the filter, but it did improve their journeys when they used it 
- Users who did not use the filter were not disrupted by it
- Users didn’t recognise that they could scroll on the filter pop-up 

**Quotes**:
Comment from one of participants who didn’t notice the filter: _"Fairly easy, I didn't like that you need to scroll all the way down to find the appointment you need"_ 
_"all I'm thinking about is making my appointments so didn't think about filter”
"it needs a little arrow on the filtering screen, to show that there is more to come"_

**Recommendations from the team:** 
- Consider making filter sticky, so that it stays on the screen as users naturally start scrolling, they would notice the filter option earlier 
- Team specific recommendations included: adding a date filter option, and removing the ‘no preference’ option

**Comment by joshavery01** [comment]
> <img alt="Screenshot 2024-04-23 at 11 41 20" width="374" src="https://private-user-images.githubusercontent.com/14331000/324799718-72997c33-5b7a-4ce8-ba89-fed571a7cdd7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg5NjE4NjcsIm5iZiI6MTc0ODk2MTU2NywicGF0aCI6Ii8xNDMzMTAwMC8zMjQ3OTk3MTgtNzI5OTdjMzMtNWI3YS00Y2U4LWJhODktZmVkNTcxYTdjZGQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjAzVDE0MzkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE3OWU3NzhjNjYwNzU3ODRkZjM3MTA2MTExNzEyMzU1NjJiNzg3NjNmM2VjZDBlYmQ1OWM2Y2Y5ODZmMGNkNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0t9s6W6zscfhCIxxfdNb4nL7HjiRtbQA5ZK1nkHWEtA">
> Notification & messaging

![Image](https://github.com/user-attachments/assets/d8d707b5-896b-459c-9c32-2eb9dee913cb)

Notifications and Messaging have updated the inbox filtering to include an option for 'flagged' messages to support the new flagging feature.

**The flagging feature:**
- Allows users to 'flag' a message, which marks it with an orange tag for better identification in a list of messages.
- Allows users to 'unflag' a previously marked message to return it to it's default state.

**Update to our filters**
By providing an option in the filters for 'flagged' messages, it allows users to further manage and customise what they can see in their inbox. 

**Upcoming challenge**
With 'remove/restore' on its way, we're exploring how to manage the filtering of senders where all messages from a particular sender have been removed from the user's inbox (essentially, where filtering by that particular option would have no results).
- Should the sender's name still appear in the filters?
- Should we include a caveat to indicate whether the filtering includes or excludes removed messages?

For now, we're keeping the sender's name in the filtering but it's a topic of discussion at the moment on N&M. Any ideas or similar examples would be really appreciated!
